### PR TITLE
DOC: examples for roots of Bessel functions

### PR DESCRIPTION
**Pull request by dschmitz89**
#### What does this implement/fix?
Provides examples for all functions listed unter [Zeros of Bessel functions](https://scipy.github.io/devdocs/reference/special.html#zeros-of-bessel-functions) except [jnjnp_zeros](https://scipy.github.io/devdocs/reference/generated/scipy.special.jnjnp_zeros.html#scipy.special.jnjnp_zeros) which I cannot really make sense of at the moment.

#### Additional information
Should not be merged until doc build is working again.

### CODE REVIEW

**Comment by dschmitz89** [comment]
Is there a recommended way of rounding long numpy arrays which are otherwise longer than then the 79 characters by the linter in an example?

File: scipy/special/_basic.py [hunk 1]
```diff
@@ -581,7 +760,44 @@ def y1p_zeros(nt, complex=False):
...
+    >>> ax.scatter(real_roots, y1_values_at_roots.real, s=30, c='k',
+    ...            label=r"Extrema of $Y_1$", zorder=5)
+    >>> ax.hlines(0, 0, xmax, color='k')
+    >>> ax.set_ylim(-0.5, 0.5)
+    >>> ax.set_xlim(0, xmax)
+    >>> ax.legend(ncol=2, bbox_to_anchor=(1., 0.75))
+    >>> plt.tight_layout()
+    >>> plt.show()
+        """
     if not isscalar(nt) or (floor(nt) != nt) or (nt <= 0):
```
Comment (tupui) [inline scipy/special/_basic.py:801]:
```suggestion
    >>> plt.show()
    """
    if not isscalar(nt) or (floor(nt) != nt) or (nt <= 0):
```

File: scipy/special/_basic.py [hunk 2]
```diff
@@ -271,6 +271,45 @@ def jnyn_zeros(n, nt):
...
+    >>> jn_roots, jnp_roots, yn_roots, ynp_roots = jnyn_zeros(1, 3)
+    >>> fig, ax = plt.subplots()
+    >>> xmax= 11
+    >>> x = np.linspace(0, xmax)
+    >>> ax.plot(x, jn(1, x), label=r"$J_1$", c='r')
+    >>> ax.plot(x, jvp(1, x, 1), label=r"$J_1'$", c='b')
+    >>> ax.plot(x, yn(1, x), label=r"$Y_1$", c='y')
+    >>> ax.plot(x, yvp(1, x, 1), label=r"$Y_1'$", c='c')
+    >>> zeros = np.zeros((3, ))
+    >>> ax.scatter(jn_roots, zeros, s=30, c='r', zorder=5,
```
Comment (tupui) [inline scipy/special/_basic.py:299]: Not requesting anything here as we don't have a policy yet on the subject. Prefer changing line/markers styling vs colours to help people with colour deficiency issues.

**Review by tupui — COMMENTED** [review]
Thanks @dschmitz89. LGTM modulo a few comments.

File: scipy/special/_basic.py [hunk 3]
```diff
@@ -303,7 +342,9 @@ def jn_zeros(n, nt):
 
     See Also
     --------
-    jv
+    jv: Real-order Bessel functions of the first kind
+    jn: Integer-order Bessel functions of the first kind
```
Comment (tupui) [inline scipy/special/_basic.py:346]: This seems to be the culprit.

Comment (dschmitz89) [reply scipy/special/_basic.py]: Thanks!

**Review by tupui — APPROVED** [review]
Thank you for the update @dschmitz89. LGTM and renders good in the doc.

**Pull request merged**